### DISCUSSION TITLE
Add zxing in visual studio

### DISCUSSION THREAD

**Discussion by NBarile** [zxing-cpp/zxing-cpp#506] [Q&A] [answered]
Hi ! 

How can I add zxing in my Visual Studio 2022 project ?

I follow the instruction to download and use cmake, but i didn't see how to use it....

Thanks !

**Comment by axxel** [comment]
Short feedback from the maintainer:
 1. I'm not using Visual Studio (or windows for that matter) myself.
 1. This is a very general question, I suggest googling for something along the lines of "use cmake project from Visual Studio".
 1. There is also the [vcpkg](https://github.com/Microsoft/vcpkg/tree/master/ports/nu-book-zxing-cpp) available. No idea how that is helpful in this context, though.

**Comment by duanqn** [comment]
I do use Visual Studio and I can share some suggestions based on my personal experience:
1. You can use CMake with Visual Studio. This does require your VS project to be a CMake project, though. This cannot be used with a traditional solution-based project.
2. As axxel mentioned above, you can use the vcpkg port of ZXing-cpp. It is up-to-date with the latest 2.0.0 release. You can take a look at https://github.com/microsoft/vcpkg#quick-start-windows, which contains instructions for using vcpkg with Visual Studio. From my understanding it supports both solution-based and CMake-based projects.

**Comment by NBarile** [comment, marked as answer]
Hi !

It works better ith cmale and vcpkg ^^ 

I follow this tutorial for VS2022
https://www.youtube.com/watch?v=FeBzSYiWkEU&ab_channel=elibro

You should use vcpkg install nu-book-zxing-cpp to install the zxing lib
And add find_package(Threads) before zxing package in cmakelist.txt
